### Q1. What are the three measures of central tendency?

In [1]:
print("Three Measures Of Central Tendecy Are : Mean , Median , Mode")

Three Measures Of Central Tendecy Are : Mean , Median , Mode


### Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

Differences :

Mean is affected by all data points, including outliers, which makes it a good measure for datasets without extreme values.

Median is less affected by outliers and skewed data, providing a better central value in such cases.

Mode identifies the most common value and is not affected by outliers, but it may not provide a sense of centrality in data with no repeating values or multiple modes.

Application : 

Mean is often used in financial data, scientific research, and performance metrics where every value matters.

Median is preferred in income distribution, real estate, and any dataset with outliers.

Mode is commonly used in market research, voting data, and other categorical data analysis.

### Q3. Measure the three measures of central tendency for the given height data:

In [2]:
import numpy as np

In [3]:
heights = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [4]:
np.mean(heights)

np.float64(177.01875)

In [5]:
np.median(heights)

np.float64(177.0)

In [7]:
import statistics

In [8]:
statistics.mode(heights)

178

In [9]:
statistics.multimode(heights)

[178, 177]

### Q4. Find the standard deviation for the given data:

In [10]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [11]:
np.std(data)

np.float64(1.7885814036548633)

### Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

1. Range : 

    Definition: The range is the simplest measure of dispersion, calculated as the difference between the maximum and minimum values in a dataset.

    Formula: Range = Maximum Value − Minimum Value

In [12]:
data = [4, 8, 6, 5, 3, 7, 9, 2]
range_value = max(data) - min(data)
print("Range:", range_value)

Range: 7


2. Variance : 

    Definition: Variance measures how much the data points deviate from the mean. It calculates the average of the squared differences between each data point and the mean.


In [16]:
data = [4, 8, 6, 5, 3, 7, 9, 2]
variance_value = np.var(data)
print("Variance:", variance_value)

Variance: 5.25


3. Standard Deviation :

    Definition: Standard deviation is the square root of the variance, bringing the measure of dispersion back to the same unit as the data

In [17]:
data = [4, 8, 6, 5, 3, 7, 9, 2]
np.std(data)


np.float64(2.29128784747792)

Example : 

Consider two datasets representing the exam scores of students in two different classes:

In [31]:
class_A = [70, 72, 68, 74, 76]
class_B = [50, 95, 60, 90, 65]


In [32]:
cls_A_avg = np.mean(class_A)
cls_B_avg = np.mean(class_B)

In [33]:
cls_A_avg

np.float64(72.0)

In [34]:
cls_B_avg

np.float64(72.0)

In [35]:
range_A = max(class_A) - min(class_A)
range_B = max(class_B) - min(class_B)

In [36]:
variance_A = np.var(class_A)
variance_B = np.var(class_B)

In [37]:
std_A = np.std(class_A)
std_B = np.std(class_B)

In [38]:
print(f"CLass A Avg : {cls_A_avg}")
print(f"CLass A Range : {range_A}")
print(f"CLass A Variance : {variance_A}")
print(f"CLass A Standard Deviation : {std_A}")
print(f"CLass B Avg : {cls_B_avg}")
print(f"CLass B Range : {range_B}")
print(f"CLass B Variance : {variance_B}")
print(f"CLass B Standard Deviation : {std_B}")

CLass A Avg : 72.0
CLass A Range : 8
CLass A Variance : 8.0
CLass A Standard Deviation : 2.8284271247461903
CLass B Avg : 72.0
CLass B Range : 45
CLass B Variance : 306.0
CLass B Standard Deviation : 17.4928556845359


### Explanation of the Results:
#### Average (Mean):

    Both Class A and Class B have the same mean score of 72.0. This indicates that, on average, students in both classes scored the same. However, this doesn't tell us anything about how the scores are distributed around this mean.

#### Range:

    Class A: The range is 8 (from 68 to 76), meaning that the scores are tightly clustered around the mean.
    
    Class B: The range is 45 (from 50 to 95), which indicates a much broader spread of scores. This suggests that while the average is the same, individual scores in Class B vary much more widely.

#### Variance:

    Class A: The variance is 8.0, which is relatively low, indicating that the scores in Class A are close to the mean.
    
    Class B: The variance is 306.0, which is much higher, showing that the scores in Class B are spread out further from the mean.

#### Standard Deviation:

    Class A: The standard deviation is 2.83, a low value that confirms the scores are close to the mean.
    
    Class B: The standard deviation is 17.49, a high value that suggests there is a lot of variability in the scores.

#### Interpretation:
    
    Class A: The low range, variance, and standard deviation indicate that students' scores are relatively consistent, clustering closely around the mean of 72. This suggests a more uniform performance among students in Class A.

    Class B: The high range, variance, and standard deviation indicate a much more varied performance among students. While the average score is the same as Class A, some students scored much higher or much lower, leading to a broader spread of scores.

#### Conclusion:

Class A has more consistent performance with scores close to the mean, while Class B has more variability, with scores widely dispersed around the mean. This suggests that while the average performance is the same, the level of consistency in Class A is much higher compared to Class B.

### Q6. What is a Venn diagram?

A Venn diagram is a visual representation used to illustrate the relationships between different sets or groups of objects.

### Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

In [39]:
A = {2,3,4,5,6,7}
B = {0,2,6,8,10}

##### (i) A intersect B

In [41]:
intersect = A.intersection(B)
print(intersect)

{2, 6}


##### (ii) A ⋃ B

In [42]:
union = A.union(B)
print(union)

{0, 2, 3, 4, 5, 6, 7, 8, 10}


### Q8. What do you understand about skewness in data?

Skewness in data refers to the asymmetry or lack of symmetry in the distribution of data values. It indicates the direction and degree to which the distribution deviates from the normal distribution, which is symmetric. Skewness helps in understanding the shape of the data distribution.

Positive Skewness (Right-Skewed) : Indicates right skewness , Mean > Median > Mode , The distribution has a longer right tail.
Example : Income distribution in many countries is often right-skewed, with most people earning a moderate amount and a few individuals earning significantly more.

Negative Skewness (Left-Skewed) : Indicates left skewness , Mean < Median < Mode , The distribution has a longer left tail.
Example : The age at retirement is often left-skewed, with most people retiring at a later age and fewer people retiring very early.

No Skewness (Symmetrical Distribution) : Indicates zero skewness , Mean = Median = Mode , The distribution has no skewness, and the tails on both sides are of equal length.
Example : A perfectly normal distribution (bell curve) has no skewness.

### Q9. If a data is right skewed then what will be the position of median with respect to mean?

If a dataset is right-skewed (positively skewed), the position of the median will be to the left of the mean.

The median, being the middle value of the dataset, is less affected by the extreme values in the tail and remains closer to the bulk of the data.

Example : 

Mean income might be higher due to a few very high earners.

Median income represents the income of the middle person and will be lower than the mean, reflecting the typical income level more accurately.

### Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

##### Covariance : 

Covariance measures the degree to which two variables change together. It indicates whether an increase in one variable generally leads to an increase (or decrease) in the other variable.

##### Correlation : 

Correlation measures both the strength and direction of a linear relationship between two variables. It is the standardized version of covariance and is dimensionless, meaning it does not depend on the units of the variables.

##### Usage In Statistical Analysis : 

Covariance is primarily used to understand the direction of the relationship between variables and is often a step in calculating other statistics, like the correlation coefficient or in multivariate statistics.

Correlation is widely used in:

Data Analysis: To determine the strength and direction of linear relationships between variables.

Financial Analysis: To understand the relationship between different financial assets (e.g., stocks).

Predictive Modeling: To identify and select features that are linearly related to the target variable.

### Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

![alt text](<Screenshot 2024-08-15 130723.png>)

In [43]:
data = [5, 8, 12, 7, 10]

mean = sum(data) / len(data)

print("Sample Mean (Basic Python):", mean)

Sample Mean (Basic Python): 8.4


### Q12. For a normal distribution data what is the relationship between its measure of central tendency?

For a normal distribution (also known as a Gaussian distribution), the measures of central tendency are all equal and perfectly aligned. This is a defining characteristic of the normal distribution. 

The mean is the average value of the dataset and is the point at which the data is centered.

The median is the value that divides the dataset into two equal halves. In a normal distribution, it coincides with the mean.

The mode is the value that occurs most frequently in the dataset. In a normal distribution, the mode also coincides with the mean.

For a normal distribution : Mean = Median = Mode

### Q13. How is covariance different from correlation?

Differences : 

Covariance depends on the units of the variables, making its magnitude hard to interpret in isolation.

Correlation is normalized and dimensionless, making it easier to understand and compare.

### Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Outliers are data points that differ significantly from other observations in a dataset. They can have a substantial impact on both measures of central tendency (mean, median, mode) and measures of dispersion (range, variance, standard deviation).

Mean : Outliers can heavily influence the mean because the mean is calculated by summing all data points and dividing by the number of points.

Median : The median is less affected by outliers because it represents the middle value of a sorted dataset. The median remains stable even if extreme values are present, as long as the dataset is sufficiently large.

Mode : Outliers generally have little impact on the mode unless they are extremely frequent. The mode is the most common value in the dataset, so a single extreme value doesn’t affect it unless it appears frequently.

Range : The range is directly affected by outliers because it is calculated as the difference between the maximum and minimum values. Outliers can inflate the range, making it larger.

Variance & Std : Both variance and standard deviation are affected by outliers because they involve the squared differences from the mean. Outliers increase these squared differences, leading to higher variance and standard deviation.

In [44]:
data1 = [10,12,14,16,18]
data1_with_outlier = [10,12,14,16,100]

In [45]:
print(f"Without Outlier : {data1}")
print(f"Mean : {np.mean(data1)}")
print(f"Median : {np.median(data1)}")
print(f"Mode : {statistics.mode(data1)}")
print(f"Range : {max(data1) - min(data1)}")
print(f"Variance : {np.var(data1)}")
print(f"Standard Deviation : {np.std(data1)}")

print("-------------------------------------------------------------------------------------------------\n")

print(f"With Outlier : {data1_with_outlier}")
print(f"Mean : {np.mean(data1_with_outlier)}")
print(f"Median : {np.median(data1_with_outlier)}")
print(f"Mode : {statistics.mode(data1_with_outlier)}")
print(f"Range : {max(data1_with_outlier) - min(data1_with_outlier)}")
print(f"Variance : {np.var(data1_with_outlier)}")
print(f"Standard Deviation : {np.std(data1_with_outlier)}")


Without Outlier : [10, 12, 14, 16, 18]
Mean : 14.0
Median : 14.0
Mode : 10
Range : 8
Variance : 8.0
Standard Deviation : 2.8284271247461903
-------------------------------------------------------------------------------------------------

With Outlier : [10, 12, 14, 16, 100]
Mean : 30.4
Median : 14.0
Mode : 10
Range : 90
Variance : 1215.0399999999997
Standard Deviation : 34.857423886454946
